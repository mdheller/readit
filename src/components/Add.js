import React, { Component } from 'react';
import { StyleSheet, View, Text, TextInput } from 'react-native';
import PropTypes from 'prop-types';

/**
 * Add component consists of the textinput field and a character counter text
 * component that is rendered in the NewScreen component.
 */
class Add extends Component {
  static defaultProps = {
    onChangeText: () => {},
  };

  constructor(props) {
    super(props);
    this.state = {
      text: '',
    };
  }

  _onChangeText(text) {
    this.setState({ text });
    this.props.onChangeText(text);
  }

  render() {
    return (
      <View style={styles.container}>
        <TextInput
          onChangeText={this._onChangeText.bind(this)}
          value={this.state.text}
          autoFocus
          multiline
        />
        <Text
          style={this.state.text.length > 255 ? styles.errorCharCounter : styles.normalCharCounter}
        >
          {this.state.text.length}/255
        </Text>
      </View>
    );
  }
}

Add.propTypes = {
  onChangeText: PropTypes.func,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
  },
  normalCharCounter: {
    color: 'black',
  },
  errorCharCounter: {
    color: 'red',
  },
});

export default Add;
