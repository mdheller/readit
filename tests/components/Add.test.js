import 'react-native';
import React from 'react';
import Add from '../../src/components/Add';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Add />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
