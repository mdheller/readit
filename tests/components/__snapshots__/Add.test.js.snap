// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoFocus={true}
    multiline={true}
    onChangeText={[Function]}
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
      }
    }
  >
    0
    /255
  </Text>
</View>
`;
